lib/twilio-ruby/http/http_client.rb

```diff
@@ -31,7 +31,11 @@ def _request(request)
           f.options.params_encoder = Faraday::FlatParamsEncoder
           f.request :url_encoded
           f.headers = request.headers
-          f.request(:basic_auth, request.auth[0], request.auth[1])
+          if Faraday::VERSION.start_with?('2.')
```

Comment:
I think that this condition should check for Faraday v1.x or later because v1.x shows the following warning:
```
WARNING: `Faraday::Connection#basic_auth` is deprecated; it will be removed in version 2.0.
While initializing your connection, use `#request(:basic_auth, ...)` instead.
See https://lostisland.github.io/faraday/middleware/authentication for more usage info.
```

How about...
```suggestion
          if Faraday::VERSION >= '1.'
```

Reply:
Where are you seeing the deprecation warning? I don't think it is coming from here. What I have here is the correct usage for versions prior to 2.0 and later. Passing `:authorization, :basic` does not work in versions <2.0 and will raise an argument error.

https://lostisland.github.io/faraday/middleware/authentication

Reply:
My mistake. I was testing it on an old version of twilio-ruby. The issue was fixed in #568. Please accept my apologies.